### PR TITLE
Renamed ICellProperties to ICellDefinition to make things clearer

### PR DESCRIPTION
**Pull request by hhyyrylainen**
where the different classes need to be used

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/early_multicellular_stage/editor/EarlyMulticellularEditor.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public override bool CanCancelAction
     public Patch? SelectedPatch => patchMapTab.SelectedPatch;
 
     [JsonIgnore]
-    public ICellProperties? EditedCellProperties => selectedCellTypeToEdit;
+    public ICellDefinition? EditedCellProperties => selectedCellTypeToEdit;
```
Comment (CI09) [inline src/early_multicellular_stage/editor/EarlyMulticellularEditor.cs:82]: EditedCellProperties could also be renamed, not necessarily though

File: src/microbe_stage/editor/ICellEditorData.cs [hunk 2]
```diff
@@ -3,5 +3,5 @@
 /// </summary>
 public interface ICellEditorData : IHexEditor, IEditorWithPatches, IEditorWithActions
 {
-    public ICellProperties? EditedCellProperties { get; }
+    public ICellDefinition? EditedCellProperties { get; }
```
Comment (CI09) [inline src/microbe_stage/editor/ICellEditorData.cs:6]: Same here

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 3]
```diff
@@ -49,7 +49,7 @@ public class MicrobeEditor : EditorBase<EditorAction, MicrobeStage>, IEditorRepo
         editedSpecies ?? throw new InvalidOperationException("species not initialized");
 
     [JsonIgnore]
-    public ICellProperties EditedCellProperties =>
+    public ICellDefinition EditedCellProperties =>
```
Comment (CI09) [inline src/microbe_stage/editor/MicrobeEditor.cs:52]: Here too

File: src/early_multicellular_stage/editor/EarlyMulticellularEditor.cs (hunk 1 above)
Comment (hhyyrylainen) [reply src/early_multicellular_stage/editor/EarlyMulticellularEditor.cs:82]: I was thinking about that, but as goes so high up the chain that at that level there doesn't really logically exist a cell definition that is being modified, but just the general properties of a thing that is being modified. I know that's not the best explanation but when I did initially rename this property as well, it didn't really fit in my head as to what the editor does, so I left it unused. This shouldn't really be able to cause confusion as there doesn't exist a gameplay simulation in the editor that would have `Components.CellProperties` that could be confused with this.

**Review by CI09 — APPROVED** [review]
Seems good but I was playing early multicellular prototype and my main cell membrane froze at red color from damage.
Also in awakening stage when founding society center, during the animationg where camera goes upwards game showed be patch extinction menu but it is likely not related

**Pull request merged**